src/solace_ai_connector_rest/components/rest_base.py

```diff
@@ -68,7 +70,8 @@ def health_check():
         self.register_routes()
 
     def run(self):
-        self.app.run(host=self.host, port=self.listen_port)
+        http_server = WSGIServer((self.host, self.listen_port), self.app)
+        http_server.serve_forever()
 
     def stop_component(self):
         func = self.app.config.get("werkzeug.server.shutdown")
```

Comment:
You'll need to update the stop_component logic. I believe you need to call http_server.stop() for a graceful shutdown, but a little research is warranted.